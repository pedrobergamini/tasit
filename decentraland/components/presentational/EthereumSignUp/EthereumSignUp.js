import React from "react";
import { StyleSheet, KeyboardAvoidingView } from "react-native";
import { responsiveHeight } from "react-native-responsive-dimensions";
import { Header } from "react-navigation";
import PropTypes from "prop-types";
import Colors from "@shared-constants/Colors";
import LargeText from "@presentational/LargeText";
import EthereumSignUpForm from "@presentational/EthereumSignUpForm";

export default class EthereumSignUp extends React.Component {
  render() {
    const OFFSET = responsiveHeight(3);
    const { onSignUp } = this.props;
    return (
      <KeyboardAvoidingView
        keyboardVerticalOffset={Header.HEIGHT + OFFSET}
        style={styles.container}
        behavior="padding"
      >
        <LargeText>{`Cool. Let's start by picking your Tasit username.`}</LargeText>
        <EthereumSignUpForm onSignUp={onSignUp} />
      </KeyboardAvoidingView>
    );
  }
}

EthereumSignUp.propTypes = {
  onSignUp: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    justifyContent: "center",
  },
});
