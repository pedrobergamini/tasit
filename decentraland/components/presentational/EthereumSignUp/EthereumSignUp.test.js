import React from "react";
import { shallow } from "enzyme";
import EthereumSignUp from "@presentational/EthereumSignUp";

describe("EthereumSignUp", () => {
  it("renders the component", async () => {
    const onSignUp = () => {};
    expect(shallow(<EthereumSignUp onSignUp={onSignUp} />)).toMatchSnapshot();
  });
});
