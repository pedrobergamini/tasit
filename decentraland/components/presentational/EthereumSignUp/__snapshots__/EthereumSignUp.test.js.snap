// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthereumSignUp renders the component 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={104.02}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LargeText>
    Cool. Let's start by picking your Tasit username.
  </LargeText>
  <EthereumSignUpForm
    onSignUp={[Function]}
  />
</KeyboardAvoidingView>
`;
